grizli/prep.py

```diff
@@ -4716,14 +4932,18 @@ def visit_grism_sky(grism={}, apply=True, column_average=True, verbose=True, ext
                         label=grism['files'][j].split('_fl')[0])
 
         # result
-        fp = open(file.replace('_flt.fits', '_column.dat'), 'wb')
+        fp = open(file.replace('_flt.fits', '_column.dat').replace('_rate.fits', '_column.dat'), 'wb')
         fp.write(b'# column obs_resid ok resid uncertainty\n')
         np.savetxt(fp, np.array([xmsk, yres, yok*1, y_pred-bg_sky, gp_sigma]).T, fmt='%.5f')
         fp.close()
 
         if apply:
             # Subtract the column average in 2D & log header keywords
-            gp_res = np.dot(y_pred[:, None]-bg_sky, np.ones((1014, 1)).T).T
+            if isJWST:
```

Comment:
Can get the required shape from `im_shape` and avoid the `if/else` statement here

Reply:
fixed this and a few other instances here: 60d97b9d92fc01949ea01c96cba3c660b9ba8478